Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -11,9 +11,17 @@ public protocol Rule {
     func validate(file: File, compilerArguments: [String]) -> [StyleViolation]
     func validate(file: File) -> [StyleViolation]
     func isEqualTo(_ rule: Rule) -> Bool
+
+    // These are called by the linter and are always implemented in extensions.
+    func collect(infoFor file: File, into storage: RuleStorage, compilerArguments: [String])
```

Comment:
Nit: I think this should be called `collectInfo(for:into:compilerArguments:)`

```suggestion
    func collectInfo(for file: File, into storage: RuleStorage, compilerArguments: [String])
```